### PR TITLE
cicd: workflow refactor

### PR DESCRIPTION
**Pull request by gomex**
the build agent shouldn't push image on PR

### CODE REVIEW

File: .github/workflows/build-agent.yml [hunk 1]
```diff
@@ -9,12 +9,6 @@ on:
     - 'agent/**'
     - '.github/workflows/build-agent.yml'
 
-  pull_request:
-    branches: [ master ]
-    paths:
-    - 'agent/**'
-    - '.github/workflows/build-agent.yml'
```
Comment (otavio) [inline .github/workflows/build-agent.yml:16]: It must be build but Docker image should not to be pushed.

File: .github/workflows/build-agent.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ on:
     paths:
     - 'agent/**'
     - '.github/workflows/build-agent.yml'
-
+
```
Comment (otavio) [inline .github/workflows/build-agent.yml:11]: Useless change.

File: .github/workflows/build-agent.yml [hunk 3]
```diff
@@ -47,21 +48,29 @@ jobs:
...
         run: echo ::set-env name=RELEASE_VERSION::${GITHUB_REF#refs/*/}
 
-      - name: build image
-        if: "!contains(github.ref, 'refs/tags/v')"
+      - name: build image 
+        if: ${{ github.event_name == 'pull_request' }} 
+        run: |
+          docker buildx build -f agent/Dockerfile.${{ matrix.dockerfile }} \
+            --tag shellhubio/agent:${{ github.sha }}-${{ matrix.dockerfile }} \
+            --platform linux/${{ matrix.arch }} .
```
Comment (otavio) [inline .github/workflows/build-agent.yml:56]: This and the below task seem the same. Aren't they?

File: .github/workflows/commit.yml [hunk 4]
```diff
@@ -3,6 +3,9 @@ on:
   push:
     branches-ignore:
       - 'dependabot/*'
+  
+  pull_request:
+    branches: [ master ]
```
Comment (otavio) [inline .github/workflows/commit.yml:8]: Please split the commit check in another PR so it can go in fast; also this PR is changing two different things which are not a good practice and we should avoid whenever possible.

File: .github/workflows/build-agent.yml (hunk 3 above)
Comment (gomex) [reply .github/workflows/build-agent.yml:56]: No, this step don't have "--push" option.

File: .github/workflows/build-agent.yml (hunk 2 above)
Comment (gomex) [reply .github/workflows/build-agent.yml]: Done

File: .github/workflows/commit.yml (hunk 4 above)
Comment (gomex) [reply .github/workflows/commit.yml]: Removed.

File: .github/workflows/build-agent.yml (hunk 3 above)
Comment (otavio) [reply .github/workflows/build-agent.yml:55]: Couldn't you have a docker push as a specific task?

Comment (gomex) [reply .github/workflows/build-agent.yml:55]: I can try, but my idea is keep it simple. The buildx execution would be responsible for everything. 
I will work to separate these steps
